### PR TITLE
Add REDUCTION_CERTIFICATE

### PR DESCRIPTION
**Pull request by blegat**
Corresponds to Mosek statuses:
* https://docs.mosek.com/8.1/capi/constants.html#mosek.solsta.prim_illposed_cer
* https://docs.mosek.com/8.1/capi/constants.html#mosek.solsta.dual_illposed_cer

Closes https://github.com/JuliaOpt/MathOptInterface.jl/issues/725

### CODE REVIEW

**Comment by blegat** [comment]
> I think you're supposed to put a hyphen in "ill-posed".

Indeed, I used illposed as this is what is used in the Mosek doc but it seems ill-posed is the correct form.

File: src/attributes.jl [hunk 1]
```diff
@@ -855,15 +855,23 @@ The values indicate how to interpret the result vector.
   infeasibility.
 * `NEARLY_INFEASIBILITY_CERTIFICATE`: the result satisfies a relaxed criterion for
   a certificate of infeasibility.
+* `REDUCTION_CERTIFICATE`: the result vector is an illposed certificate. If the
```
Comment (mlubin) [inline src/attributes.jl]: Agreed on "ill-posed" over "illposed".

Comment (mlubin) [inline src/attributes.jl]: This is a pretty esoteric status. Could you add a link to a page/paper that describes what it means?

File: src/attributes.jl [hunk 2]
```diff
@@ -855,15 +855,24 @@ The values indicate how to interpret the result vector.
   infeasibility.
 * `NEARLY_INFEASIBILITY_CERTIFICATE`: the result satisfies a relaxed criterion for
   a certificate of infeasibility.
+* `REDUCTION_CERTIFICATE`: the result vector is an ill-posed certificate; see
+  [this article](https://doi.org/10.1007/s10107-017-1169-9) for details. If the
```
Comment (mlubin) [inline src/attributes.jl]: It's better to link to the freely available version: https://arxiv.org/abs/1408.4685.

**Pull request merged**